runtime/cudaq/platform/fermioniq/helpers/FermioniqServerHelper.cpp

```diff
@@ -0,0 +1,512 @@
+  bool jobIsDone(ServerMessage &getJobResponse) override;
+
+  /// @brief Processes the server's response to a job retrieval request and
+  /// maps the results back to sample results.
+  cudaq::sample_result processResults(ServerMessage &postJobResponse,
+                                      std::string &jobId) override;
+
+  void refreshTokens(bool force_refresh);
+
+#if 0
+  /// @brief Update `passPipeline` with architecture-specific pass options
+  void updatePassPipeline(const std::filesystem::path &platformPath,
+                          std::string &passPipeline) override;
+#endif
+
+  /// @brief Return next results polling interval
+  std::chrono::microseconds
+  nextResultPollingInterval(ServerMessage &postResponse) override;
+
+private:
+  /// @brief RestClient used for HTTP requests.
+  // RestClient client;
+
+  /// @brief API Key for Fermioniq API
+  std::string token;
+
+  /// @brief user_id of logged in user
+  std::string userId;
+
+  std::vector<std::string> circuit_names;
+
+  /// @brief exp time of token
+  std::chrono::system_clock::time_point tokenExpTime;
+
+  /// @brief Helper method to retrieve the value of an environment variable.
+  std::string getEnvVar(const std::string &key, const std::string &defaultVal,
+                        const bool isRequired) const;
+
+  /// @brief Helper function to get value from config or return a default value.
+  std::string getValueOrDefault(const BackendConfig &config,
+                                const std::string &key,
+                                const std::string &defaultValue) const;
+
+  /// @brief Helper method to check if a key exists in the configuration.
+  bool keyExists(const std::string &key) const;
+};
+
+// Initialize the Fermioniq server helper with a given backend configuration
+void FermioniqServerHelper::initialize(BackendConfig config) {
+  cudaq::info("Initializing Fermioniq Backend.");
+
+  parseConfigForCommonParams(config);
+
+  backendConfig[CFG_URL_KEY] =
+      getEnvVar("FERMIONIQ_API_BASE_URL", DEFAULT_URL, false);
+  backendConfig[CFG_API_KEY_KEY] =
+      getEnvVar("FERMIONIQ_API_KEY", DEFAULT_API_KEY, false);
+
+  backendConfig[CFG_ACCESS_TOKEN_ID_KEY] =
+      getEnvVar("FERMIONIQ_ACCESS_TOKEN_ID", "", true);
+  backendConfig[CFG_ACCESS_TOKEN_SECRET_KEY] =
+      getEnvVar("FERMIONIQ_ACCESS_TOKEN_SECRET", "", true);
+
+  backendConfig[CFG_USER_AGENT_KEY] =
+      "cudaq/" + std::string(cudaq::getVersion());
+
+  if (config.find("project_id") != config.end()) {
+    backendConfig[CFG_PROJECT_ID_KEY] = config.at("project_id");
+  }
+
+  if (config.find("remote_config") != config.end()) {
+    backendConfig[CFG_REMOTE_CONFIG_KEY] = config["remote_config"];
+  }
+  if (config.find("noise_model") != config.end()) {
+    backendConfig[CFG_NOISE_MODEL_KEY] = config["noise_model"];
+  }
+
+  if (config.find("bond_dim") != config.end()) {
+    backendConfig[CFG_BOND_DIM_KEY] = config.at("bond_dim");
+  }
+
+  refreshTokens(true);
+}
```

Comment:
Would you mind adding a "Fermioniq" section in https://github.com/NVIDIA/cuda-quantum/blob/main/docs/sphinx/using/backends/hardware.rst to describe these configurations? Thanks!